src/stave.js

```diff
@@ -319,14 +298,96 @@ Vex.Flow.Stave = (function() {
 
     addEndTimeSignature: function(timeSpec, customPadding) {
       this.addEndModifier(new Vex.Flow.TimeSignature(timeSpec, customPadding));
+      return this;
     },
 
     addTrebleGlyph: function() {
-      this.clef = "treble";
-      this.addGlyph(new Vex.Flow.Glyph("v83", 40));
+      this.addClef('treble');
       return this;
     },
 
+    getModifiers: function(position) {
+      if (position === undefined) return this.modifiers;
+
+      return this.modifiers.filter(function(modifier) {
+        return modifier.getPosition() === position;
+      });
+    },
+
+    sortByCategory: function(items, order) {
+      for (var i = items.length - 1; i >= 0; i--) {
+        for (var j = 0; j < i; j++) {
+          if (order[items[j].getCategory()] > order[items[j + 1].getCategory()]) {
+            var temp = items[j];
+            items[j] = items[j + 1];
+            items[j + 1] = temp;
+          }
+        }
+      }
+    },
+
+    format: function() {
```

Comment:
Sounds nice! I agree with you. Some kind of removing work is better for the future.

I added commits of `set` function implementations and also added a new test case.

This test case also verify working of changing begin & end stavemodifier after drawing.
<img width="521" alt="screen shot 2016-01-22 at 5 41 02 pm" src="https://cloud.githubusercontent.com/assets/2025065/12506067/8650d972-c12f-11e5-9365-d2989b957d14.png">

There exists too many test cases which use `add` because some functions like `addClef` or `addTimeSignature` are very essential so used in many other kinds of test cases. How to make deprecated might be better to have a discussion more.